### PR TITLE
One Clojure lint to rule them all

### PR DESCRIPTION
**Pull request by bshepherdson**
**Background**

We had two lint configs:
- `lint-config.edn` which was enfored by Github Actions
- `.clj-kondo/config.edn`, the default location used by editors

**Problem**

Patchy enforcement of the (larger) config in `.clj-kondo/config.edn` since not
everyone uses an editor.

Some new lint, like canonical namespace aliases (#21738, #19930) is not enforced
automatically and has begun to bit-rot.

**Solution**

Combine both configs, fix any outstanding lint, and update our tooling to use
the unified config.

**Caveats**

Anyone who has their local editor configured to use `lint-config.edn` will have
their linting broken by this change.

### CODE REVIEW

File: docs/developers-guide/devenv.md [hunk 1]
```diff
@@ -214,7 +214,7 @@ clojure -X:dev:ee:ee-dev:drivers:drivers-dev:eastwood
 clojure -X:dev:ee:ee-dev:drivers:drivers-dev:test:namespace-checker
 
 # Run clj-kondo
-clj-kondo --parallel --lint src shared/src enterprise/backend/src --config lint-config.edn
+clj-kondo --parallel --lint src shared/src enterprise/backend/src --config .clj-kondo/config.edn
```
Comment (dpsutton) [inline docs/developers-guide/devenv.md:217]: the config is in the default location so you can omit `--config .clj-kondo/config.edn` if you like

File: src/metabase/api/tiles.clj [hunk 2]
```diff
@@ -70,7 +70,7 @@
 
 ;;; --------------------------------------------------- RENDERING ----------------------------------------------------
 
-(defn- ^BufferedImage create-tile [zoom points]
+(defn- create-tile ^BufferedImage [zoom points]
```
Comment (dpsutton) [inline src/metabase/api/tiles.clj:73]: 🍾 i think we sponsored this linter for exactly this type of situation.

File: test/metabase/api/session_test.clj [hunk 3]
```diff
@@ -170,7 +170,7 @@
                                               {"x-forwarded-for" "10.1.2.3"})
               status-code (:status response)]
           (assert (= status-code 401) (str "Unexpected response status code:" status-code))))
-      (dotimes [n 50]
+      (dotimes [n 40]
```
Comment (dpsutton) [inline test/metabase/api/session_test.clj:173]: what's the purpose of this change?

Comment (bshepherdson) [reply test/metabase/api/session_test.clj:173]: The delay was too long, and it was failing the test (especially with coverage instrumentation). The resulting error string was "wait 9 more seconds" but the regex wanted "wait 1\d more seconds". (Gross, I know.)

**Review by dpsutton — APPROVED** [review]
